main.py

```diff
@@ -1,63 +1,87 @@
-import random
-from time import time
+import random, curses
+import time
 from dico import DICO
 
-random.seed(time())
+stdscr = curses.initscr()
+stdscr.nodelay(True)
+random.seed(time.time())
 
-def get_input_player(toto):
-  inp = raw_input("Choose a letter (not in %s)" % toto)
-  letter = inp[0].strip()
-  if letter == '':
-    print("Please choose a letter")
-    return get_input_player()
-  toto+= letter
-  return letter
+def get_input_player(tried):
+  inp = stdscr.getch()
+  if inp == -1:
+    return None
+  if chr(inp) in tried:
+    return get_input_player(tried)
+  return chr(inp)
 
 def is_in_word(word, letter):
   if letter in word:
-    print("We have to print all %s" % letter)
     return True
   return False
 
+def turn(word, tried, buffer, lives):
+  play = True
+  inp = get_input_player(tried)
+  if inp is None:
+    return play, lives
+  tried.append(inp)
+  if is_in_word(word, inp):
+    for i in range(len(word)):
+      if word[i] == inp:
+        buffer[i] = inp
+  else:
+    lives -= 1 
+  play = lives > 0
+  if not "_" in buffer or lives <= 0:
+    play = False
+  return play, lives
+    
+
+def print_data(buffer, lives, tried):
+  stdscr.addstr("Player: ")
+  stdscr.addstr("%s lives\n" % lives, curses.A_UNDERLINE)
+  stdscr.addstr("%s\n" % buffer)
+  stdscr.addstr("Tried Letters: %s\n" %tried)
+
+def print_endgame(word, win=False):
+  if win:
+    stdscr.addstr("You Won ! Good Job !")
+  else:
+    stdscr.addstr("You Lost ! Sorry.")
+  stdscr.addstr("The Word was ")
+  stdscr.addstr(word, curses.A_UNDERLINE)
+  while None == get_input_player([]):
```

Comment:
That is weird, could we set get_input_player to have an empty list by default so we don't have to pass it.

Reply:
dsadsadsa